machine/v1beta1/types_awsprovider.go

```diff
@@ -160,6 +165,28 @@ type SpotMarketOptions struct {
 	MaxPrice *string `json:"maxPrice,omitempty"`
 }
 
+type InstanceMetadataHTTPTokens string
+
+const (
+	// InstanceMetadataHTTPTokensRequired enforces sending of a signed token header with any instance metadata retrieval (GET) requests.
+	// Enforces IMDSv2 usage.
+	InstanceMetadataHTTPTokensRequired = "required"
+	// InstanceMetadataHTTPTokensOptional allows IMDSv1 usage along with IMDSv2
+	InstanceMetadataHTTPTokensOptional = "optional"
+)
+
+// MetadataServiceOptions defines the options available to a user when configuring
+// Instance Metadata Service (IMDS) Options.
+type MetadataServiceOptions struct {
+	// HTTPTokens determines whether or not the host requires the use of authentication when interacting with the metadata service.
+	// When using authentication, this enforces v2 interaction method (IMDSv2) with the metadata service.
+	// When omitted, this means the user has no opinion and the value is left to the platform to choose a good
+	// default, which is subject to change over time. The current default is optional.
+	// +kubebuilder:validation:Enum=required;optional
+	// +optional
+	HTTPTokens InstanceMetadataHTTPTokens `json:"httpTokens,omitempty"`
```

Comment:
This looks like an `authenticationMode`.  We are developing a kube-like API for cluster-admins to interact with.  Not trying to expose an AWS API via the kube-apiserver.

Reply:
>Not trying to expose an AWS API via the kube-apiserver.

Since this whole resource is AWS specific and literally is exposure of AWS api capabilities to cluster-admin, i want to keep it as is.

https://github.com/openshift/enhancements/pull/1075#discussion_r839682923 - prior discussion

Reply:
> Since this whole resource is AWS specific and literally is exposure of AWS api capabilities to cluster-admin, i want to keep it as is.

It is AWS specific, but it exposed as surface area via the kube-apiserver and should be consistent with kube-like standards.

Reply:
IMHO - it must be consistent with logic within single resource first of all.